include/secp256k1_schnorr.h

```diff
@@ -68,104 +77,123 @@ SECP256K1_API int secp256k1_schnorr_recover(
+ *  Out: stage2sig64:        pointer to a 64-byte array to store the signature
+ *  In:  other_stage1sig32s: pointer to an array of num_others pointers to
+ *                           32-byte stage 1 partial multisignatures from all
+ *                           other cosigners (can only be NULL if num_others is
+ *                           0)
+ *       num_others:         the number of cosigners (excluding yourself)
+ *       msg32:              the 32-byte message hash being signed (cannot be
+ *                           NULL)
+ *       sec32:              pointer to a 32-byte secret key (cannot be NULL)
+ *       noncefp:            pointer to a nonce generation function. If NULL,
+ *                           secp256k1_nonce_function_default is used
+ *       ndata:              pointer to arbitrary data used by the nonce
+ *                           generation function (can be NULL)
  *
- * The intended procedure for creating a multiparty signature is:
- * - Each signer S[i] with private key x[i] and public key Q[i] runs
- *   secp256k1_schnorr_generate_nonce_pair to produce a pair (k[i],R[i]) of
- *   private/public nonces.
- * - All signers communicate their public nonces to each other (revealing your
- *   private nonce can lead to discovery of your private key, so it should be
- *   considered secret).
- * - All signers combine all the public nonces they received (excluding their
- *   own) using secp256k1_ec_pubkey_combine to obtain an
- *   Rall[i] = sum(R[0..i-1,i+1..n]).
- * - All signers produce a partial signature using
- *   secp256k1_schnorr_partial_sign, passing in their own private key x[i],
- *   their own private nonce k[i], and the sum of the others' public nonces
- *   Rall[i].
- * - All signers communicate their partial signatures to each other.
- * - Someone combines all partial signatures using
- *   secp256k1_schnorr_partial_combine, to obtain a full signature.
- * - The resulting signature is validatable using secp256k1_schnorr_verify, with
- *   public key equal to the result of secp256k1_ec_pubkey_combine of the
- *   signers' public keys (sum(Q[0..n])).
+ *  The second stage uses the stage 1 partial signatures from all other
+ *  cosigners and computes a stage 2 partial signature. If num_others is 0, the
+ *  result is a full signature (though different than the one produced by
+ *  secp256k1_schnorr_sign, given the same msg32, sec32, noncefp, ndata).
  *
- *  Note that secp256k1_schnorr_partial_combine and secp256k1_ec_pubkey_combine
- *  function take their arguments in any order, and it is possible to
- *  pre-combine several inputs already with one call, and add more inputs later
- *  by calling the function again (they are commutative and associative).
+ *  All cosigners must use the same msg32, and the same as in stage1. You must
+ *  also use the same noncefp/ndata for your own stage1 and stage2. Other
+ *  participants may use different nonce generation, though.
```

Comment:
I think you should be more explicit: "Different participants may use different nonce generating functions and data, as long as they are each consistent between stage 1 and stage 2."

Up to you.